main/src/library/TcpSocket.flix

```diff
@@ -14,9 +14,52 @@
  *  limitations under the License.
  */
 
+import java.io.InputStream
+import java.io.OutputStream
 import java.net.Socket
 
 ///
 /// Represents a TCP socket
 ///
-enum TcpSocket(Socket)
+enum TcpSocket(Socket, InputStream, OutputStream)
+
+instance Reader[TcpSocket] {
+
+    type Aef = IO
+
+    pub def read(
```

Comment:
I think this can just be one line. Also I dont think the line breaks are needed. Keep it short and sweet.

(Same comment applies elsewhere)